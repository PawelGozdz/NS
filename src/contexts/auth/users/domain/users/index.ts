export * from './errors';
export * from './events';
export * from './user.aggregate-root';
export * from './user.aggregate-root.fixture';
export * from './user.snapshot';
export * from './users-command-repository.interface';
export * from './users-query-repository.interface';
