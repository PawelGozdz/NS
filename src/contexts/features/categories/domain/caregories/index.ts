export * from './category-command-repository.interface';
export * from './category-query-repository.interface';
export * from './category.aggregate-root';
export * from './category.aggregate-root.fixture';
export * from './category.snapshot';
export * from './errors';
export * from './events';
