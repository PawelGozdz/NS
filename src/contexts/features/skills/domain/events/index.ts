export * from './skill-created.event';
