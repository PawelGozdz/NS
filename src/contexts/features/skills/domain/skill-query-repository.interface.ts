import { ISkillsQueryParams } from '@app/core';

export type SkillInfo = {
  id: number;
  name: string;
  context: string;
  parentId: number | null;
  description: string | null;
};

export abstract class ISkillsQueryRepository {
  abstract getManyBy(query: ISkillsQueryParams): Promise<SkillInfo[]>;
}
