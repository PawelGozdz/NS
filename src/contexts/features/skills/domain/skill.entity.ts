export class Skill {
  id: number;

  name: string;

  description?: string | null;

  parentId?: number | null;

  context: string;

  constructor(props: Skill) {
    this.id = props.id;
    this.name = props.name;
    this.description = props.description ?? null;
    this.parentId = props.parentId ?? null;
    this.context = props.context;
  }

  public static create({
    id,
    name,
    description,
    parentId,
    context,
  }: {
    id: number;
    name: string;
    description?: string | null;
    parentId?: number | null;
    context: string;
  }): Skill {
    const skill = new Skill({
      id,
      name,
      description: description ?? null,
      parentId: parentId ?? null,
      context,
    });

    return skill;
  }
}

export type ISkillCreateData = {
  name: string;
  description?: string | null;
  parentId?: number | null;
  context: string;
};

export type ISkillUpdateData = {
  id: number;
  name?: string;
  description?: string | null;
  context?: string;
  parentId?: number | null;
};
