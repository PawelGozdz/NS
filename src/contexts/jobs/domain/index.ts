export * from './job-position';
export * from './job-user-profile';
export * from './jobs';
export * from './skill';
