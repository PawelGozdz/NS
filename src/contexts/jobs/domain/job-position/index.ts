export * from './errors';
export * from './events';
export * from './job-position-command-repository.interface';
export * from './job-position-entity.fixture';
export * from './job-position-query-repository.interface';
export * from './job-position.entity';
export * from './job-position.snapshot';
