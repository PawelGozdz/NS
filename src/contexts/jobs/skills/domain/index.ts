export * from './errors';
export * from './events';
export * from './skill-command-repository.interface';
export * from './skill-entity.fixture';
export * from './skill-query-repository.interface';
export * from './skill.entity';
