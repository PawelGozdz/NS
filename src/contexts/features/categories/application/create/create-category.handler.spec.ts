import { createMock } from '@golevelup/ts-jest';
import { Test } from '@nestjs/testing';

import { TestCqrsModule, TestLoggerModule, catchActError } from '@libs/testing';

import { CategoryAlreadyExistsError, CategoryEntityFixtureFactory, ICategoriesCommandRepository } from '../../domain';
import { CreateCategoryCommand } from './create-category.command';
import { CreateCategoryHandler } from './create-category.handler';

describe('CreateCategoryHandler', () => {
  let handler: CreateCategoryHandler;
  let categoryCommandRepositoryMock: jest.Mocked<ICategoriesCommandRepository>;

  beforeEach(async () => {
    categoryCommandRepositoryMock = createMock();

    const app = await Test.createTestingModule({
      imports: [TestLoggerModule.forRoot(), TestCqrsModule],
      providers: [
        CreateCategoryHandler,
        {
          provide: ICategoriesCommandRepository,
          useValue: categoryCommandRepositoryMock,
        },
      ],
    }).compile();

    handler = app.get(CreateCategoryHandler);
  });

  const name = 'test';
  const context = 'test-context';
  const id = 1;
  const command = new CreateCategoryCommand({
    name,
    context,
  });

  const categoryEntity = CategoryEntityFixtureFactory.create();

  describe('Success', () => {
    it('should create new category', async () => {
      // Arrange
      categoryCommandRepositoryMock.getOneByNameAndContext.mockResolvedValueOnce(undefined);
      categoryCommandRepositoryMock.save.mockResolvedValueOnce({ id });

      // Act
      const result = await handler.execute(command);

      // Assert

      expect(categoryCommandRepositoryMock.save).toHaveBeenCalledWith({
        name,
        context,
        description: undefined,
        parentId: undefined,
      });
      expect(result).toMatchSnapshot();
    });
  });

  describe('failure', () => {
    it('should throw CategoryAlreadyExistsError', async () => {
      // Arrange
      categoryCommandRepositoryMock.getOneByNameAndContext.mockResolvedValueOnce(categoryEntity);

      // Act
      const { error } = await catchActError(() => handler.execute(command));

      // Assert

      expect(categoryCommandRepositoryMock.save).toHaveBeenCalledTimes(0);
      expect(error).toBeInstanceOf(CategoryAlreadyExistsError);
      expect(error).toMatchSnapshot();
    });
  });
});
