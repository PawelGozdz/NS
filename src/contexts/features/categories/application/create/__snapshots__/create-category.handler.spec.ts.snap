// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCategoryHandler Success should create new category 1`] = `
{
  "id": 1,
}
`;

exports[`CreateCategoryHandler failure should throw CategoryAlreadyExistsError 1`] = `[CategoryAlreadyExistsError: Entity with name test already exists]`;
